### PR TITLE
feat(truffle-examples): adds `caller-pays-for-query` example

### PR DESCRIPTION
**Pull request by gskapka**
...per title. Tests passing, files linted (`.sol` & `.js`).

### CODE REVIEW

File: solidity/truffle-examples/caller-pays-for-query/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      ✓ User cannot make query if msg.value < query cost
+      ✓ User can make query if msg.value === query cost
+      ✓ Query should have emitted event with ETH price in USD (11676ms)
+      ✓ Eth price in USD should be saved in contract
+      ✓ User should get refund when making query but sending > query cost (121ms)
+
+
+  10 passing (12s)
+
+truffle(develop)>
```
Comment (riccardopersiani) [inline solidity/truffle-examples/caller-pays-for-query/README.md]:
```suggestion
```

Was this left on purpose to show that testing happen inside truffle develop?

File: solidity/truffle-examples/caller-pays-for-query/package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "oraclize-examples-using-truffle__computation-datasource-kraken-price-ticker",
```
Comment (riccardopersiani) [inline solidity/truffle-examples/caller-pays-for-query/package.json]:
```suggestion
  "name": "oraclize-examples-using-truffle__url-datasource-caller-pays-for-query",
```

File: solidity/truffle-examples/caller-pays-for-query/package.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  "version": "0.1.0",
+  "description": "Testing set up for Oraclize's Kraken-Price-Ticker example contract",
```
Comment (riccardopersiani) [inline solidity/truffle-examples/caller-pays-for-query/package.json]:
```suggestion
  "description": "Testing set up for Provable's Caller-Pays-For-Query example contract",
```

**Comment by riccardopersiani** [comment]
Example tested successfully, all test passed

**Comment by gskapka** [comment]
> Is this reliable enough for inclusion in the ethereum-api CI?

I don't see why not - it's using the same query as Kraken and the test accounts always have a million ETH so I'd say it ought to be included.

That way the `provable_setCustomGasPrice` API method also comes under the CI umbrella, which is nice!

File: solidity/truffle-examples/caller-pays-for-query/README.md (hunk 1 above)
Comment (gskapka) [reply solidity/truffle-examples/caller-pays-for-query/README.md]: Hahah no - much less thought than that, just whacked the paste in and didn't check it!

**Pull request merged**